lib/Sema/CSSyntacticElement.cpp

```diff
@@ -1281,9 +1255,18 @@ class SyntacticElementConstraintGenerator
     }
 
     auto contextualResultInfo = getContextualResultInfo();
+
+    // Single-expression closures need their contextual type locator anchored
+    // on the closure itself.
+    if (context.isSingleExpressionClosure(cs) && returnStmt->hasResult()) {
```

Comment:
Can this be a part of `getContextualResultInfo()` so we don't have to mutate it post factum?

Reply:
Good idea! How's this?